docs/release-notes/charmcraft-4.0.rst

```diff
@@ -0,0 +1,171 @@
+
+Charmcraft is an important member of the Canonical community and ecosystem, and all its
+site names should demonstrate that. With this release, we gave the docs a new home at
+the Ubuntu domain. You can now reach it at:
+
+`documentation.ubuntu.com/charmcraft <https://documentation.ubuntu.com/charmcraft>`_
+
+We put redirects in place to handle links to the old ReadTheDocs domain.
+
+With this change, we also removed the language subdirectory (/en) in the URL, to
+shave a few characters off all links.
+
+OpenID Connect integration
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+OpenID Connect is now available in all 12-factor app charm extensions.
+It's available in any charm using the
+:ref:`Django <django-framework-extension>`,
+:ref:`Express <expressjs-framework-extension>`,
+:ref:`FastAPI <fastapi-framework-extension>`,
+:ref:`Flask <flask-framework-extension>`, and
+:ref:`Go <go-framework-extension>` extensions and packed with Charmcraft 4.
+
+``init`` profile changes
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Charmcraft 4.0 has much more modern :ref:`ref_commands_init` ``machine`` and
+``kubernetes`` profiles. Changes include:
+
+- The ``static`` and ``lint`` tox environments are now united.
+- The charm is packed with the :ref:`craft_parts_uv_plugin`.
+- Dependencies are now managed with `uv <https://docs.astral.sh/uv>`_.
+- The reusable integration tests now use `Jubilant
+  <https://documentation.ubuntu.com/jubilant/>`_.
+
+Test profiles
+^^^^^^^^^^^^^
+
+We introduced two new init profiles, ``test-machine`` and ``test-kubernetes``. They
+provide a framework for using the :ref:`charmcraft test <ref_commands_test>` command.
+These profiles are experimental, so we can't guarantee their stability.
+
+Spring Boot framework extension
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Apps that use the Spring Boot framework can now use the
+:ref:`spring-boot-framework-extension` for charming the app. A new
+:ref:`tutorial <write-your-first-kubernetes-charm-for-a-spring-boot-app>` is also
+included to walk a first-time charmer through charming their Spring Boot app.
+
+Feature removals
+----------------
+
+The following features are removed in Charmcraft 4.0. If you need these features, they
+are still available in Charmcraft 3.
+
+Windows support
+~~~~~~~~~~~~~~~
+
+Charmcraft 4.0 has ceased support for Windows. To continue on the platform, we
+encourage Windows users to run Charmcraft in `Windows Subsystem for Linux
+<https://ubuntu.com/desktop/wsl>`_.
+
+
+CentOS 7 support
+~~~~~~~~~~~~~~~~
+
+CentOS 7 reached its `end of life
+<https://www.redhat.com/en/topics/linux/centos-linux-eol>`_ on 30 June 2024, and
+support for it has ended with Charmcraft 4.0.
+
+
+Bundle removal
+~~~~~~~~~~~~~~
+
+Aligning with the `discontinuation of bundle registrations
+<https://discourse.charmhub.io/t/15344>`_, Charmcraft 4.0 removes all bundle-related
+functionality. Charmcraft 3 will remain supported for packing bundles as well as the
+bundle commands (``register-bundle`` and ``promote-bundle``).
```

Comment:
Sorry for the back and forth on this one, but I saw some ways we could be more attentive.

```suggestion
As scheduled, all charm bundle features are removed with Charmcraft 4.0. Charmcraft 3
will continue support for packing bundles and the ``register-bundle`` and
``promote-bundle`` commands. This follows the `discontinuation of new bundle
registrations <https://discourse.charmhub.io/t/15344>`_ in Nov 2024.
```